### ISSUE TITLE
Compile failure on FreeBSD

### ISSUE THREAD

**Issue by xanderio** [alexrsagen/rs-nfc1-sys#1]
This crate is unable to be build on the FreeBSD platform as the linux i2c and spi headers aren't found. 

```
  In file included from /usr/home/xanderio/Projects/rs-nfc1-sys/vendor/nfc/libnfc/drivers/pn532_i2c.c:52:
  /usr/home/xanderio/Projects/rs-nfc1-sys/vendor/nfc/libnfc/buses/i2c.h:43:12: fatal error: 'linux/i2c-dev.h' file not found
  #  include <linux/i2c-dev.h>
             ^~~~~~~~~~~~~~~~~
  1 error generated.
  gmake[2]: *** [libnfc/CMakeFiles/nfc.dir/build.make:230: libnfc/CMakeFiles/nfc.dir/drivers/pn532_i2c.c.o] Error 1
  gmake[2]: *** Waiting for unfinished jobs....
  In file included from /usr/home/xanderio/Projects/rs-nfc1-sys/vendor/nfc/libnfc/drivers/pn532_spi.c:48:
  /usr/home/xanderio/Projects/rs-nfc1-sys/vendor/nfc/libnfc/buses/spi.h:43:12: fatal error: 'linux/spi/spidev.h' file not found
  #  include <linux/spi/spidev.h>
             ^~~~~~~~~~~~~~~~~~~~
  1 error generated.
  gmake[2]: *** [libnfc/CMakeFiles/nfc.dir/build.make:244: libnfc/CMakeFiles/nfc.dir/drivers/pn532_spi.c.o] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:131: libnfc/CMakeFiles/nfc.dir/all] Error 2
  gmake: *** [Makefile:156: all] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2
```

**Comment by xanderio** [comment]
I have identified the problem and have a working patch. I'll try to upstream it, or should we also apply it to the forked version, that this crate uses?

**Comment by xanderio** [comment]
https://github.com/nfc-tools/libnfc/pull/652

**Comment by alexrsagen** [comment]
I see that nfc-tools/libnfc#652 got merged, that's great. I will pull the latest changes from libnfc to my fork, so that FreeBSD builds will work 😄 

My reason for maintaining a fork is that I made some changes to fix MSVC builds in alexrsagen/libnfc which may potentially be breaking. As such, I don't know if it's a great idea to upstream my changes... If you know any libnfc maintainers, feel free to have them take a look at the two latest commits in my fork and maybe we can see about upstreaming those.

In any case, `nfc1` and `nfc1-sys` will be updated later today.
